data/abi/abi_encode.go

```diff
@@ -478,6 +479,13 @@ func decodeTuple(encoded []byte, childT []Type) ([]interface{}, error) {
 	return values, nil
 }
 
+// maxAppArgs is the maximum number of arguments for an application call transaction
```

Comment:
Let's put a comment in explaining that we have this as a constant here in order to obey ARC4, else one would expect to get it from consensus parameters. But (I suppose) if we do update the arg max to, say, 32, ARC4 apps would still want to encode this way, despite the change.